File: scope104/src/main/java/com/petro/scope104/data/network/GitHubService.java
package com.petro.scope104.data.network;

import com.petro.scope104.data.network.response.UserListResponse;

import io.reactivex.rxjava3.core.Observable;
import retrofit2.http.GET;
import retrofit2.http.Query;

public interface GitHubService {
    @GET("api")
    Observable<UserListResponse> listRepos(
            @Query("nat") String nationality,
            @Query("page") int page,
            @Query("results") int limit,
            @Query("gender") String gender
    );
}

File: scope104/src/main/kotlin/com/petro/scope104/data/network/GitHubService.kt
package com.petro.scope104.data.network

import com.petro.scope104.data.network.response.UserListResponse
import io.reactivex.rxjava3.core.Observable
import retrofit2.http.GET
import retrofit2.http.Query

interface GitHubService {
    @GET("api")
    fun listRepos(
        @Query("nat") nationality: String?,
        @Query("page") page: Int,
        @Query("results") limit: Int,
        @Query("gender") gender: String?
    ): Observable<UserListResponse>
}